index.js

```diff
@@ -10,31 +10,26 @@ fs.readdir('./frames').then(data => {
   data.forEach(async frame => {
     const f = await fs.readFile(`./frames/${frame}`);
     frames.push(f.toString());
-  })
+  });
 });
 
-const colorsOptions = ['red', 'yellow', 'green', 'blue', 'magenta', 'cyan', 'white'];
-const numColors = colorsOptions.length;
+const selectColor = function(previousColor) {
+  const colors = ['red', 'yellow', 'green', 'blue', 'magenta', 'cyan', 'white'];
+  const availableColors = colors.filter(color => color !== previousColor);
```

Comment:
This seems wasteful. Why go through the loop each time when we really only care about collisions?

```js
do {
  result = availableColors[Math.floor(Math.random() * availableColors.length)]
} while (color !== previousColor);

return result;
```